shotgun_api3/shotgun.py

```diff
@@ -424,7 +426,8 @@ def __init__(self,
                  password=None,
                  sudo_as_login=None,
                  session_token=None,
-                 auth_token=None):
+                 auth_token=None,
+                 rpc_attempt_interval=None):
```

Comment:
the constructor already has a gazillion paramters. We should not add any more unless they are considered high frequency use ones. Can this be a parameter instead? e.g.

```
sg = Shotgun(site_name, script_name, script_key)
sg.rpc_attempt_interval = 1000 # adjusting default interval
```

Unless we have a strong reason for using the constructor approach, i think we should go with the approach above. Thoughts?

Reply:
My main worry with a parameter on the instance is that it goes against the grain of the current api. I'm not saying passing in a parameter is necessarily the better approach either to be honest.

As long as it can be documented properly, I think setting it as a parameter on the instance is fine. It's not a long lasting value like the host name or the credentials.

Reply:
I think the only change I'd make to your proposal Manne is to make it so that
`sg.config.rpc_attempt_internal` is there the value gets set, as the config object is where we keep settings.

Reply:
The pattern exists already - see for example:
https://groups.google.com/a/shotgunsoftware.com/forum/#!msg/shotgun-dev/P-YXTWs9mp0/tzXtkIALt1sJ

Yes yes! I totally agree that we should keep it on the `config` object!

Reply:
I've moved `rpc_attempt_interval` back to being just a property of `_Config`, and improved the comment there, after discussing with @manneohrstrom and @thebeeland how best to document this.